pyiron_atomistics/atomistics/structure/neighbors.py

```diff
@@ -230,19 +282,18 @@ def _get_distances_and_indices(
             )
         self._extended_indices = indices.copy()
         indices[distances<np.inf] = self._get_wrapped_indices()[indices[distances<np.inf]]
-        if allow_ragged is None:
-            allow_ragged = self.allow_ragged
-        if allow_ragged:
-            return self._contract(distances, distances), self._contract(indices, distances)
-        return distances, indices
+        return self._reshape(distances, mode, distances), self._reshape(indices, mode, distances)
 
+    @deprecate(allow_ragged="use mode instead.")
     def get_indices(
         self,
         positions=None,
         allow_ragged=None,
+        mode='filled',
         num_neighbors=None,
         cutoff_radius=np.inf,
         width_buffer=1.2,
+        flatten=False,
```

Comment:
flatten is not used here!

Reply:
Good job! Before I introduced `mode` I was thinking about using `flatten`. I simply forgot to remove it.